lsp-iedit.el

```diff
@@ -73,8 +74,29 @@ language server doesn't support renaming.
 See also `lsp-enable-symbol-highlighting'."
   (interactive)
   (let ((highlights (lsp-request "textDocument/documentHighlight"
-                                 (lsp--text-document-position-params))))
-    (lsp-iedit--on-ranges (mapcar #'lsp:document-highlight-range highlights))))
+                                 (lsp--text-document-position-params)))
+        (-compare-fn (lambda (a b)
+                       (let* ((a-range (gethash "range" a))
```

Comment:
you should not use gethash with structures coming from the server since a might be plist when lsp-use-plist is t. You should use either `lsp-range-*` or &Range destructuring. Als, there is `lsp--position-equal`.

Reply:
Havent tested but something like that should do:

``` elisp
(-lambda ((&Range :start l-start :end l-end) (&Range :start r-start :end r-end))
  (and (lsp--position-equal l-start l-end)
       (lsp--position-equal r-start r-end)))
```